### PR TITLE
Add byexample

### PR DESCRIPTION
**Pull request by saraedum**

### CODE REVIEW

**Comment by saraedum** [comment]
@eldipa: Please ping me here if you would like to be a maintainer of this conda-forge package as well.

File: recipes/byexample/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - python
+
+test:
+  commands:
+    - byexample --version
+
+about:
+  home: http://github.com/byexamples/byexample
+  license: GPL-3.0
+  license_file: LICENSE
```
Comment (marcelotrevisani) [inline recipes/byexample/meta.yaml]: The ``license`` is not being packaged on pypi. Could you please raise an issue about it on this project?
For now, you can package it manually, please see
https://conda-forge.org/docs/maintainer/adding_pkgs.html#packaging-the-license-manually

Comment (saraedum) [reply recipes/byexample/meta.yaml]: Thanks, I reported it upstream and added a LICENSE to the recipe.

**Comment by saraedum** [comment]
Needs https://github.com/conda-forge/staged-recipes/pull/8998

**Comment by saraedum** [comment]
Ok. I guess I have to wait for a bit for pyte to make it through the CDN.

**Comment by saraedum** [comment]
@conda-forge/help-python @marcelotrevisani I think this recipe is also ready for review :)

**Comment by saraedum** [comment]
Can this be noarch if it ships a Python script?

**Comment by marcelotrevisani** [comment]
> Can this be noarch if it ships a Python script?

If it is just ``entry_points`` you can add them to the recipe and it should be fine
However, if it is platform-specific, it cannot be ``noarch``

File: recipes/byexample/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 44cc2bd00141e3e7f9ccf82eff025385c58ece02cd36da04bea2f0937d2410bb
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (marcelotrevisani) [inline recipes/byexample/meta.yaml]: Please add
```suggestion
  script: "{{ PYTHON }} -m pip install . -vv"
  entry_points:
    - byexample = byexample.byexample:main
```

**Comment by saraedum** [comment]
I see, wasn't clear to me since the documentation said `Similar to a regular conda package, a noarch Python package will contain a site-packages and info directory that define the package. In addition to the standard files in the info directory there is link.json that defines the type of noarch package and any entry points or scripts. Entry points, defined in the setup.py style entry_points['console_scripts'] will be created by conda when the package is installed`.
I hadn't understood that I have to add the entry points again explicitly.

**Comment by marcelotrevisani** [comment]
It is failing because you are using the ``fcntl`` which is platform specific.
So, this recipe cannot be ``noarch``

**Comment by marcelotrevisani** [comment]
Indeed, you can remove the ``entry_points`` then

File: recipes/byexample/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 44cc2bd00141e3e7f9ccf82eff025385c58ece02cd36da04bea2f0937d2410bb
+
+build:
+  noarch: python
```
Comment (marcelotrevisani) [inline recipes/byexample/meta.yaml]:
```suggestion
  skip: True  # [win]
```

**Comment by marcelotrevisani** [comment]
``fcntl``: https://docs.python.org/3/library/fcntl.html

> This module performs file control and I/O control on file descriptors. It is an interface to the fcntl() and ioctl() **Unix routines**.

**Comment by saraedum** [comment]
@marcelotrevisani thanks. I was not aware of this. I'll open an issue upstream as I don't think that that's intentional.

**Comment by saraedum** [comment]
@marcelotrevisani I assume that the `entry_points` don't hurt?

**Comment by saraedum** [comment]
> @marcelotrevisani thanks. I was not aware of this. I'll open an issue upstream as I don't think that that's intentional.

Indeed, all it does is lock on a file. I am sure there's some more portable way of doing this. But that's for a future release.

**Comment by saraedum** [comment]
@marcelotrevisani finally it seems to work :)

**Comment by eldipa** [comment]
> @eldipa: Please ping me here if you would like to be a maintainer of this conda-forge package as well.

Thanks, but I have zero knowledge about `conda` so the "maintainer" status would be too much for me. Do take this in the wrong way, if you have any problems just open an issue (like you did) and I will do my best to fix it.

**Pull request merged**